examples/pages/Categorized.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+import React, { Component } from 'react';
+import { withValue } from 'react-value';
+
+import { groupedOptions } from '../data';
+
+import { Li, Ul, Div } from '../../src/primitives';
+import { Link, H1 } from '../components';
+
+import { components as SelectComponents } from '../../src';
+
+import Select from '../../src';
+const SelectWithValue = withValue(Select);
+
+class Group extends Component<*, *> {
+  allOptionsSelected = () => {
+    const { selectValue, isOptionSelected } = this.props;
+    return this.children()
+      .map(i => isOptionSelected(i, selectValue))
+      .every(isTrue => isTrue);
+  };
+  children() {
+    const { group, menuOptions: { render } } = this.props;
+    const selectedGroup = render.find(i => i.key == group.key);
+    return selectedGroup ? selectedGroup.children : [];
+  }
+  valuesIncludingChildren = () => {
+    const { selectValue, isOptionSelected } = this.props;
+    return [
+      ...this.children()
+        .filter(i => !isOptionSelected(i, selectValue))
+        .map(i => i.data),
+      ...selectValue,
+    ];
+  };
+  valuesExcludingChildren = () => {
+    const { selectValue, isOptionSelected } = this.props;
+    return selectValue
+      .filter(i => !isOptionSelected(i, this.children()))
+      .map(i => i.data);
+  };
+  toggleGroup = () => () => {
+    const { selectValues } = this.props;
+
+    if (this.allOptionsSelected()) {
+      selectValues(this.valuesExcludingChildren());
+    } else {
+      selectValues(this.valuesIncludingChildren());
+    }
+  };
+  render() {
+    const {
+      children,
+      components,
+      getStyles,
+      innerProps,
+      isOptionSelected,
+      label,
+      menuOptions,
+      selectValue,
+      selectValues,
+      ...others
+    } = this.props;
+    const { Heading } = components;
+    return (
+      <Li
+        aria-label={label}
+        css={getStyles('group', this.props)}
+        {...innerProps}
+        {...others}
+      >
+        <Heading getStyles={getStyles} onClick={this.toggleGroup()}>
+          <input type="checkbox" checked={this.allOptionsSelected()} readOnly />
+          {label}
+        </Heading>
+        <Ul>{children}</Ul>
+      </Li>
+    );
+  }
+}
+
+const Option = props => (
+  <Div style={{ paddingLeft: '2em' }}>
```

Comment:
I wrapped the `<Option>` in  a `<Div>` and because I am not sure how I can merge new styles into the `<Option>`'s props..

Reply:
actually, I can't do that.. The following compiles but it wipes out the initial styles of the component instead of adding to them:

```
<SelectComponents.Option
      {...props}
      innerProps={{
        ...props.innerProps,
        css: { ...props.innerProps.css, paddingLeft: '2em' },
      }}
    >
```

Reply:
Ahh, I found the answer in the `Experimental.js` file..

```
const getOptionStyles = defaultStyles => ({
  ...defaultStyles,
  display: 'inline-block',
  width: '12%',
  margin: '0 1%',
  textAlign: 'center',
  borderRadius: '4px',
});
```

and then:
```
...
const defaultStyles = getStyles('option', props);
const styles = getOptionStyles(defaultStyles);
...
```

then

```
<Span {...cleanProps} css={styles} ref={innerRef}>
```